includes/functions/content-sync.php

```diff
@@ -311,3 +315,60 @@ function get_post_data( $post ) {
 	 */
 	return apply_filters( 'sophi_post_data', $data );
 }
+
+/**
+ * Check if the current data has been already tracked recently and should be skipped.
+ *
+ * @since 1.3.0
+ *
+ * @param array $data Data being tracked
+ *
+ * @return boolean
+ */
+function maybe_skip_track_event( $data ) {
+	// Ignore modifiedAt.
+	unset( $data['modifiedAt'] );
+
+	$transient_hash = substr( wp_hash( wp_json_encode( $data ) ), 0, 8 );
+	$transient_name = 'sophi_tracking_request_' . $transient_hash;
+	$tracked_data   = get_transient( $transient_name );
+
+	if ( isset( $_GET['meta-box-loader'] ) && $_GET['meta-box-loader'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		// Skip if metabox is reloading.
+		$skip = true;
+	} elseif ( $tracked_data === $data ) {
+		// Skip duplicated track within last 10 seconds.
+		$skip = true;
```

Comment:
```suggestion

	// phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
	if ( isset( $_GET['meta-box-loader'] ) && $_GET['meta-box-loader'] ) {
		// Skip if metabox is reloading.
		$skip = true;
	} elseif ( get_transient( $transient_name ) === $data ) {
		// Skip duplicated track within last 10 seconds.
		$skip = true;
```

Two suggestions:
* phpcs:ignore can go on the line above, it makes it easier to read (if the comment is on it's own line, the next line is ignored)
* I moved `get_transient()` to the `elseif` so the DB query is only made when required